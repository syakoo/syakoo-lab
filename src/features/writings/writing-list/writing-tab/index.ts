export { WritingTab } from "./writing-tab";
