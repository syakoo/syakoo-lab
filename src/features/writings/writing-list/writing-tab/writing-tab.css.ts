import { recipe } from "@vanilla-extract/recipes";

import { theme } from "@/shared/design-system/theme.css";

export const writingTabStyle = {
  item: recipe({
    base: {
      padding: theme.space[100],
      textDecoration: "none",
    },

    variants: {
      active: {
        true: {
          borderBottom: "4px solid currentColor",
          color: theme.color.text.primary,
        },
        false: {
          color: theme.color.text.secondary,
        },
      },
    },
  }),
};
