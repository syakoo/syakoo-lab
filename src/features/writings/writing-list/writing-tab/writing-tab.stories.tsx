import type { Meta, StoryObj } from "@storybook/nextjs";

import { WritingTab } from ".";

const meta = {
  component: WritingTab,
  parameters: { testLevel: "snapshot" },
} satisfies Meta<typeof WritingTab>;

export default meta;

type Story = StoryObj<typeof meta>;
export const Sample: Story = {
  args: {
    selectedType: "all",
  },
};
