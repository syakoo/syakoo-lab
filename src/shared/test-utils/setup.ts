import { exampleMDXComponent } from "@/features/mdx/mocks/fixture-mdxcomponent";

import { loadEnv } from "./nextjs/env";

// NOTE: `@mdx-js/mdx` の esm がうまく解決できなかったためまとめてモック化
// これが原因でおかしくなっていたらごめんなさい
jest.mock("@mdx-js/mdx", () => {
  return {
    runSync: jest
      .fn()
      .mockImplementation(() => ({ default: exampleMDXComponent })),
  };
});

// NOTE: `mermaid` の esm がうまく解決できなかったためまとめてモック化
// これが原因でおかしくなっていたらごめんなさい
jest.mock("mermaid", () => ({
  initialize: () => {
    return;
  },
  run: () => {
    return;
  },
}));

loadEnv();

jest.spyOn(Math, "random").mockImplementation(() => 0.5866245602626035);
