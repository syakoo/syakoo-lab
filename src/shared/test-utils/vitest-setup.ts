import { vi } from "vitest";

import { exampleMDXComponent } from "@/features/mdx/mocks/fixture-mdxcomponent";

import { loadEnv } from "./nextjs/env";

// NOTE: Next.jsのImageコンポーネントをテスト用にモック
vi.mock("next/image", () => ({
  default: "img",
}));

// NOTE: `@mdx-js/mdx` の esm がうまく解決できなかったためまとめてモック化
// これが原因でおかしくなっていたらごめんなさい
vi.mock("@mdx-js/mdx", () => {
  return {
    runSync: vi
      .fn()
      .mockImplementation(() => ({ default: exampleMDXComponent })),
  };
});

// NOTE: `mermaid` の esm がうまく解決できなかったためまとめてモック化
// これが原因でおかしくなっていたらごめんなさい
vi.mock("mermaid", () => ({
  default: {
    initialize: () => {
      return;
    },
    run: () => {
      return;
    },
  },
  initialize: () => {
    return;
  },
  run: () => {
    return;
  },
}));

loadEnv();

vi.spyOn(Math, "random").mockImplementation(() => 0.5866245602626035);
