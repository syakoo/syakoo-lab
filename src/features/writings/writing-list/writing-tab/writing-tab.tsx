import {
  type WritingListType,
  writingListTypePath,
} from "@/features/writings/writing-list/_shared/writing-list-type";
import { Row } from "@/shared/design-system/layout";
import { Link } from "@/shared/design-system/ui";

import { writingTabStyle } from "./writing-tab.css";

type WritingTabProps = {
  selectedType: WritingListType;
};

export const WritingTab: React.FC<WritingTabProps> = ({ selectedType }) => {
  const writingListTypes = [
    {
      type: "all",
      label: "All",
    },
    {
      type: "article",
      label: "Article",
    },
    {
      type: "note",
      label: "Note",
    },
    {
      type: "diary",
      label: "Diary",
    },
  ] as const satisfies readonly {
    type: WritingListType;
    label: string;
  }[];

  return (
    <Row>
      {writingListTypes.map(({ label, type }) => {
        const isActive = selectedType === type;

        return (
          <Link
            key={type}
            href={writingListTypePath(type)}
            noHovered={isActive}
          >
            <div
              aria-current={isActive}
              className={writingTabStyle.item({
                active: isActive,
              })}
            >
              {label}
            </div>
          </Link>
        );
      })}
    </Row>
  );
};
