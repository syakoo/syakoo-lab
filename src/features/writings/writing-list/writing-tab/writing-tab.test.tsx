import { testStories } from "@/shared/test-utils/test-stories";

import * as stories from "./writing-tab.stories";

describe("WritingTab", () => {
  testStories(stories);
});
