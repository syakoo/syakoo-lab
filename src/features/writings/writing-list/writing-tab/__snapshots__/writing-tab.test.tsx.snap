// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WritingTab > スナップショット Sample 1`] = `
<body>
  <div>
    <div
      class="flex_flexStyle__zqca9ti flex_flexStyle_align_stretch__zqca9tu flex_flexStyle_variants_align_stretch__zqca9t9 flex_flexStyle_direction_row__zqca9tj flex_flexStyle_gap_100__zqca9ty flex_flexStyle_variants_gap_100__zqca9td flex_flexStyle_justify_flexStart__zqca9tl flex_flexStyle_variants_justify_flexStart__zqca9t0"
    >
      <a
        class="link_linkStyle__1vz8ftz0 link_linkStyle_noHovered_true__1vz8ftz3"
        href="/writings"
      >
        <div
          aria-current="true"
          class="writing-tab_writingTabStyle_item__1g56vq60 writing-tab_writingTabStyle_item_active_true__1g56vq61"
        >
          All
        </div>
      </a>
      <a
        class="link_linkStyle__1vz8ftz0"
        href="/writings?type=article"
      >
        <div
          aria-current="false"
          class="writing-tab_writingTabStyle_item__1g56vq60 writing-tab_writingTabStyle_item_active_false__1g56vq62"
        >
          Article
        </div>
      </a>
      <a
        class="link_linkStyle__1vz8ftz0"
        href="/writings?type=note"
      >
        <div
          aria-current="false"
          class="writing-tab_writingTabStyle_item__1g56vq60 writing-tab_writingTabStyle_item_active_false__1g56vq62"
        >
          Note
        </div>
      </a>
      <a
        class="link_linkStyle__1vz8ftz0"
        href="/writings?type=diary"
      >
        <div
          aria-current="false"
          class="writing-tab_writingTabStyle_item__1g56vq60 writing-tab_writingTabStyle_item_active_false__1g56vq62"
        >
          Diary
        </div>
      </a>
    </div>
  </div>
</body>
`;
